pcl_ros/src/transforms.cpp

```diff
@@ -55,22 +69,25 @@ transformPointCloud (const std::string &target_frame, const sensor_msgs::PointCl
   }
 
   // Get the TF transform
-  tf::StampedTransform transform;
+  geometry_msgs::msg::TransformStamped transform_msg;
   try
   {
-    tf_listener.lookupTransform (target_frame, in.header.frame_id, in.header.stamp, transform);
+    transform_msg = tf_buffer.lookupTransform (target_frame, in.header.frame_id, tf2_ros::fromMsg(in.header.stamp));
   }
-  catch (tf::LookupException &e)
+  catch (tf2::LookupException &e)
   {
-    ROS_ERROR ("%s", e.what ());
+    RCLCPP_ERROR (rclcpp::get_logger("pcl_ros"), "%s", e.what ());
     return (false);
   }
-  catch (tf::ExtrapolationException &e)
+  catch (tf2::ExtrapolationException &e)
   {
-    ROS_ERROR ("%s", e.what ());
+    RCLCPP_ERROR (rclcpp::get_logger("pcl_ros"), "%s", e.what ());
     return (false);
   }
 
+  tf2::Transform transform;
+  tf2::convert (transform_msg.transform, transform);
+
```

Comment:
Same discussion points here as in transforms.hpp -- should we provide an overload of `transformAsMatrix` which accepts a `const geometry_msgs::msg::Transform&` instead of a `tf2::Transform`?

Reply:
Or change to just geometry_msgs if TF2 doesn't give back out TF2 types (which... seems odd but OK)

Reply:
Agreed it's quite odd since TF2 claims to be ROS independent. But it seems to be the case unless I'm missing something...I spent quite a while searching. Here's the API in the tf2 core library:

https://github.com/ros/geometry2/blob/noetic-devel/tf2/include/tf2/buffer_core.h